packages/frontend/src/components/FunnelChart/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { useDisclosure } from '@mantine/hooks';
+import { IconFilterOff } from '@tabler/icons-react';
+import { type ECElementEvent } from 'echarts';
+import EChartsReact from 'echarts-for-react';
+import { type EChartsReactProps, type Opts } from 'echarts-for-react/lib/types';
+import { memo, useCallback, useEffect, useState, type FC } from 'react';
+import useEchartsFunnelConfig, {
+    type FunnelSeriesDataPoint,
+} from '../../hooks/echarts/useEchartsFunnelConfig';
+import { useApp } from '../../providers/AppProvider';
+import SuboptimalState from '../common/SuboptimalState/SuboptimalState';
+import { useVisualizationContext } from '../LightdashVisualization/VisualizationProvider';
+import FunnelChartContextMenu, {
+    type FunnelChartContextMenuProps,
+} from './FunnelChartContextMenu';
+
+const EmptyChart = () => (
+    <div style={{ height: '100%', width: '100%', padding: '50px 0' }}>
+        <SuboptimalState
+            title="No data available"
+            description="Query metrics and dimensions with results."
+            icon={IconFilterOff}
+        />
+    </div>
+);
+
+const LoadingChart = () => (
+    <div style={{ height: '100%', width: '100%', padding: '50px 0' }}>
+        <SuboptimalState
+            title="Loading chart"
+            loading
+            className="loading_chart"
```

Comment:
I believe this is for scheduled deliveries?

Reply:
The classname? Does that mean it should come out?

Reply:
Correct, this is for scheduled deliveries. We should keep it so we can keep track of the loading spinners